interop-java/jvm/src/test/scala/scalaz/zio/interop/javaconcurrent.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio.interop
+
+import java.util.concurrent.{ CompletableFuture, CompletionException, CompletionStage, Future }
+
+import scalaz.zio._
+import scalaz.zio.blocking.{ blocking, Blocking }
+
+import scala.concurrent.ExecutionException
+
+object javaconcurrent {
+
+  implicit class IOObjJavaconcurrentOps(private val ioObj: IO.type) extends AnyVal {
+
+    private def unsafeCompletionStageToIO[A](cs: CompletionStage[A]): Task[A] =
+      IO.effectAsync { cb =>
+        val _ = cs.handle[Unit] { (v: A, t: Throwable) =>
+          if (v != null) {
+            cb(IO.succeed(v))
+          } else {
+            val io = t match {
+              case e: CompletionException =>
+                IO.fail(e.getCause)
+              case t: Throwable =>
+                IO.fail(t)
+            }
+            cb(io)
+          }
+        }
+      }
+
+    def fromCompletionStage[A, E >: Throwable](csIo: IO[E, CompletionStage[A]]): IO[E, A] =
```

Comment:
I think the type bound is not helping. Because of the `CompletionStage`, you will always end up with `Throwable`. `E >: Throwable` says `E` is a supertype of throwable, but the only supertype of `Throwable` is `Any` which is not useful. 

So I'd change the `csIO` to have type `Task[CompletionStage[A]]`.